cocos/platform/apple/CCFileUtils-apple.mm

```diff
@@ -60,7 +60,8 @@ static void addItemToArray(id item, ValueVector& array)
         const char* numType = [num objCType];
         if(num == (void*)kCFBooleanFalse || num == (void*)kCFBooleanTrue)
         {
-            array.push_back(Value([num boolValue]));
+            bool v = [num boolValue];
+            array.push_back(Value(v));
```

Comment:
Yep, you are right.
As mentioned [here](http://stackoverflow.com/questions/541289/objective-c-bool-vs-bool), `BOOL` is defined like this:

```
/// Type to represent a boolean value.
#if (TARGET_OS_IPHONE && __LP64__)  ||  TARGET_OS_WATCH
#define OBJC_BOOL_IS_BOOL 1
typedef bool BOOL;
#else
#define OBJC_BOOL_IS_CHAR 1
typedef signed char BOOL; 
// BOOL is explicitly signed so @encode(BOOL) == "c" rather than "C" 
// even if -funsigned-char is used.
#endif
```

Which means:
> on 64-bit iOS devices and on WatchOS `BOOL` is exactly the same thing as bool while on all other devices (OS X, 32-bit iOS) it is `signed char` and cannot even be overridden by compiler flag `-funsigned-char`

so i think we should convert it to bool explicitly.